src/abstractdataframe/subset.jl

```diff
@@ -95,17 +95,22 @@ end
            ungroup::Bool=true)
 
 Return a copy of data frame `df` or parent of `gdf` containing only rows for
-which all values produced by transformation(s) `args` for a given row are `true`.
-All transformations must produce vectors containing `true` or `false` (and
-optionally `missing` if `skipmissing=true`).
-
-Each argument passed in `args` can be either a single column selector or a
-`source_columns => function` transformation specifier following the rules
-described for [`select`](@ref).
-
-Note that as opposed to [`filter`](@ref) the `subset` function works on whole
-columns (and selects rows within groups for `GroupedDataFrame`
-rather than whole groups) and must return a vector.
+which all values produced by transformation(s) `args` for a given row are
+`true`. All transformations must produce vectors containing `true` or `false`
+(and optionally `missing` if `skipmissing=true`).
+
+Each argument passed in `args` can be any specifier following the rules
+described for [`select`](@ref) with the restriction that:
+* specifying target column name is not allowed as `subset` does not create new
+  columns;
+* every passed transformation must return a scalar or a vector (returning
+  multi-column values is not supported).
```

Comment:
This only excludes `AsTable`, right? Mentioning it would be useful, as "scalar or vector" seems to include all types one may think of.

Reply:
This excludes: `AbstractDataFrame`, `NamedTuple`, `DataFrameRow`, `AbstractMatrix`. I will change the description to reflect this.